pkg/controllers/controllers.go

```diff
@@ -106,7 +107,13 @@ func Register(ctx context.Context, systemNamespace, controllerName string, cfg c
 		klog.Infof("Starting helm controller in namespace %s", systemNamespace)
 	}
 
-	leader.RunOrDie(ctx, systemNamespace, "helm-controller-lock", appCtx.K8s, func(ctx context.Context) {
+	controllerLockName := "helm-controller-lock"
+	if controllerName != "helm-controller" {
+		klog.Infof("Starting helm controller using alias `%s`", controllerName)
+		controllerLockName = strings.Join([]string{controllerName, controllerLockName}, "-")
```

Comment:
just to be clear, this is only necessary if you're running multiple in the same namespace, right? I guess it was envisioned that the controllers would run in separate namespaces, and watch whatever namespace they're deployed to. It sounds like you're trying to deploy multiple controllers to the same ns, while watching other namespaces?

Reply:
>this is only necessary if you're running multiple in the same namespace, right?
I would clarify this to say: "multiple in the same scope".

Since PrometheusFederator exists (by default) as a cluster-scoped controller. However it sets a `managedBy` annotation matching its name, which means any controller named `helm-controller` will never deploy those charts.

And similarly the PrometheusFederator controller will never get the lock because an embedded k3s helm-controller will always get the lock first.

The example manifest is just a (admittedly contrived) example to try and show that dynamic without needing to directly involve PrometheusFederator images/charts in testing/fixing this.